Web/Edubase.Web.UIUnitTests/Areas/Governors/Controllers/GovernorControllerTests.cs

```diff
@@ -584,43 +584,76 @@ public static IEnumerable<object[]> PairwiseGovernanceProfessionalRolesAllowedCo
         }
 
         public static IEnumerable<object[]> PairwiseGovernanceProfessionalRolesExceptABC => PairwiseGovernanceProfessionalRoles
-            .Where(allPairsPair => !PairwiseGovernanceProfessionalRolesAllowedCombinations.Any(innerPair =>
+            .Where(allPairsPair =>
+                !PairwiseGovernanceProfessionalRolesAllowedCombinations.Any(innerPair =>
                     allPairsPair[0].Equals(innerPair[0])
                     && allPairsPair[1].Equals(innerPair[1]))
+                    && (eLookupGovernorRole) allPairsPair[0] != eLookupGovernorRole.GovernanceProfessionalToAnIndividualAcademyOrFreeSchool
+                    && (eLookupGovernorRole) allPairsPair[1] != eLookupGovernorRole.GovernanceProfessionalToAnIndividualAcademyOrFreeSchool
+                    && (eLookupGovernorRole) allPairsPair[0] != eLookupGovernorRole.GovernanceProfessionalToASat
+                    && (eLookupGovernorRole) allPairsPair[1] != eLookupGovernorRole.GovernanceProfessionalToASat
```

Comment:
Rather than doing the exceptions here, modify `PairwiseGovernanceProfessionalRolesAllowedCombinations` (around line 573) to specify which pairs are permitted 👍

Reply:
Something like this, perhaps:

```csharp

        /// <summary>
        /// The base rule is that governance professional roles may not co-exist.
        /// Some exceptions exist, where certain combinations are allowed - these are described below.
        /// Note that these are also enforced within the backend API, so should be kept synchronised.
        /// </summary>
        public static IEnumerable<object[]> PairwiseGovernanceProfessionalRolesAllowedCombinations
        {
            get
            {
                var allData = new List<object[]>
                {
                    //  - #198772 / #193913 : MAT can have "one-each" of "Shared governance professional - group" and "Governance professional to a multi-academy trust (MAT)"
                    new object[] {eLookupGovernorRole.Group_SharedGovernanceProfessional, eLookupGovernorRole.GovernanceProfessionalToAMat},
                    new object[] {eLookupGovernorRole.GovernanceProfessionalToAMat, eLookupGovernorRole.Group_SharedGovernanceProfessional},

                    //  - #198772 / #197361 : Establishment within a SAT can have "one-each" of "Governance professional to an individual academy or free school" and "Governance professional for single-academy trust (SAT)"
                    new object[] {eLookupGovernorRole.GovernanceProfessionalToAnIndividualAcademyOrFreeSchool, eLookupGovernorRole.GovernanceProfessionalToASat},
                    new object[] {eLookupGovernorRole.GovernanceProfessionalToASat, eLookupGovernorRole.GovernanceProfessionalToAnIndividualAcademyOrFreeSchool},

                    //  - #198239: System should allow adding a Governance professional to a federation if a record for Governance professional to a local authority maintained is already recorded.
                    new object[] {eLookupGovernorRole.GovernanceProfessionalToAFederation, eLookupGovernorRole.GovernanceProfessionalToALocalAuthorityMaintainedSchool},
                    new object[] {eLookupGovernorRole.GovernanceProfessionalToALocalAuthorityMaintainedSchool, eLookupGovernorRole.GovernanceProfessionalToAFederation},
                };
                return allData;
            }
        }
```